qutip/solver/propagator.py

```diff
@@ -82,29 +90,93 @@ def propagator(
         tlist = [0, t]
         list_output = False
     else:
-        tlist = t
+        tlist = list(t)
         list_output = True
 
     if not isinstance(H, (Qobj, QobjEvo)):
         H = QobjEvo(H, args=args, **kwargs)
 
     if isinstance(c_ops, list):
         c_ops = [QobjEvo(op, args=args, **kwargs) for op in c_ops]
+    elif c_ops is not None and not isinstance(c_ops, QobjEvo):
+        c_ops = [QobjEvo(c_ops, args=args, **kwargs)]
+
+    c_ops_list: list[QobjEvo] = []
+    if isinstance(c_ops, QobjEvo):
+        c_ops_list = [c_ops]
+    elif isinstance(c_ops, list):
+        c_ops_list = c_ops
+
+    constant_c_ops = all(op.isconstant for op in c_ops_list)
+    H_isconstant = isinstance(H, QobjEvo) and H.isconstant
```

Comment:
At  this point `H` is `QobjEvo` or `Qobj`. If it's `Qobj`, it's constant. Otherwise, it has `isconstant` attribute. So, it should be
```suggestion
H_isconstant = isinstance(H, Qobj) or H.isconstant
```